modules/database/test/std/rec/biTest.c

```diff
@@ -0,0 +1,115 @@
+    testdbPutFieldOk("test_bi_rec.INP", DBF_STRING, "test_bi_link_rec.VAL");
+    testdbPutFieldOk("test_bi_link_rec.FLNK", DBF_STRING, "test_bi_rec");
+
+    /* set VAL to on linked record */
+    testdbPutFieldOk("test_bi_link_rec.VAL", DBF_SHORT, TRUE);
+
+    /* verify that this record VAL is updated but RVAL is not */
+    testdbGetFieldEqual("test_bi_rec.VAL", DBF_SHORT, TRUE);
+    testdbGetFieldEqual("test_bi_rec.RVAL", DBF_SHORT, FALSE);
+
+    // number of tests = 6
+}
+
+static void test_raw_soft_input(void){
+    /* set soft channel */
+    testdbPutFieldOk("test_bi_rec.DTYP", DBF_STRING, "Raw Soft Channel");
+    testdbPutFieldOk("test_bi_rec.INP", DBF_STRING, "test_bi_link_rec.VAL");
+    testdbPutFieldOk("test_bi_link_rec.FLNK", DBF_STRING, "test_bi_rec");
+
+    /* set VAL to on linked record */
+    testdbPutFieldOk("test_bi_link_rec.VAL", DBF_SHORT, TRUE);
+
+    /* verify that this record RVAL and VAL are updated */
+    testdbGetFieldEqual("test_bi_rec.VAL", DBF_SHORT, TRUE);
+    testdbGetFieldEqual("test_bi_rec.RVAL", DBF_SHORT, TRUE);
+
+    // number of tests = 6
+}
+
+static void test_operator_display(void){
+    /* set operator display parameters */
+    testdbPutFieldOk("test_bi_rec.ZNAM", DBF_STRING, "ZNAM_TEST");
+    testdbPutFieldOk("test_bi_rec.ONAM", DBF_STRING, "ONAM_TEST");
+    testdbPutFieldOk("test_bi_rec.DESC", DBF_STRING, "DESC_TEST");
+
+    /* verify operator display parameters */
+    testdbGetFieldEqual("test_bi_rec.ZNAM", DBF_STRING, "ZNAM_TEST");   
+    testdbGetFieldEqual("test_bi_rec.ONAM", DBF_STRING, "ONAM_TEST"); 
+    testdbGetFieldEqual("test_bi_rec.NAME", DBF_STRING, "test_bi_rec");
+    testdbGetFieldEqual("test_bi_rec.DESC", DBF_STRING, "DESC_TEST");
+
+    // number of tests = 7
+}
+
+static void test_alarm(void){
+    /* set soft channel */
+    testdbPutFieldOk("test_bi_rec.DTYP", DBF_STRING, "Soft Channel");
+    testdbPutFieldOk("test_bi_rec.INP", DBF_STRING, "test_bi_link_rec.VAL");
+    testdbPutFieldOk("test_bi_link_rec.FLNK", DBF_STRING, "test_bi_rec");
+
+    /* Set start VAL to FALSE*/
+    testdbPutFieldOk("test_bi_link_rec.VAL", DBF_SHORT, FALSE);
+
+    /* set alarm parameters */
+    testdbPutFieldOk("test_bi_rec.ZSV", DBF_SHORT, menuAlarmSevrNO_ALARM);
+    testdbPutFieldOk("test_bi_rec.OSV", DBF_SHORT, menuAlarmSevrMINOR);
+    testdbPutFieldOk("test_bi_rec.COSV", DBF_SHORT, menuAlarmSevrINVALID);
+    
+    /* Verify alarm status is NO_ALARM*/
+    testdbGetFieldEqual("test_bi_rec.SEVR", DBF_SHORT, menuAlarmSevrNO_ALARM);
+
+    /* Set ZSV to MAJOR and verify that SEVR is now MAJOR */
+    testdbPutFieldOk("test_bi_rec.ZSV", DBF_SHORT, menuAlarmSevrMAJOR);
+    testdbGetFieldEqual("test_bi_rec.SEVR", DBF_SHORT, menuAlarmSevrMAJOR);
+
+    /* Set VAL to 1 on lined record and verify that COSV now sets the SEVR to INVALID */
+    testdbPutFieldOk("test_bi_link_rec.VAL", DBF_SHORT, TRUE);
+    testdbGetFieldEqual("test_bi_rec.SEVR", DBF_SHORT, menuAlarmSevrINVALID);
```

Comment:
You can also include here a test to check that `LALM` field holds the last value that caused the alarm, in this case it should be `1`

Reply:
Resolved in https://github.com/epics-base/epics-base/pull/340/commits/28ecf0af3c3f2585f1571215c8b16a15ccbcab37